### PR TITLE
docs: Create Concepts Docs page and Diagram

### LINKED ISSUE #3802: Concepts page

### PR DESCRIPTION
**Pull request by missBerg**
**What this PR does / why we need it**:
Includes a Concept page for the relevant resources for the docs site, with an accompanying diagram to show the relationship between the resources.

The page is supposed to help people understand the relationship between the resources.

**Which issue(s) this PR fixes**:
Fixes #3802

### CODE REVIEW

**Comment by arkodg** [comment]
thanks for adding the diagram @missBerg ! It speaks a thousand words 
one comment - the `EnvoyProxy` resource is not `required` and is optional

File: site/content/en/latest/concepts/concepts-overview.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+### SecurityPolicy
+Defines security-related policies such as authentication, authorization, and encryption settings for traffic handled by Envoy Gateway.
+
+### BackendTLSPolicy
+Defines TLS settings for backend connections, including certificate management, TLS version settings, and other security configurations. This policy is applied to Kubernetes Services.
+
+## Customize & Extend Envoy
+
+### EnvoyGatewayPatchPolicy
```
Comment (Xunzhuo) [inline site/content/en/latest/concepts/concepts-overview.md:70]: EnvoyGatewayPatchPolicy  -> EnvoyPatchPolicy

**Review by Xunzhuo — COMMENTED** [review]
Nice diagram! Very helpful, thanks @missBerg !

**Comment by missBerg** [comment]
Created a new draft: https://github.com/envoyproxy/gateway/pull/3808 
because I messed up with signing the commits.

**Pull request closed without merging**